edx_when/api.py

```diff
@@ -100,8 +119,12 @@ def clear_dates_for_course(course_key, keep=None):
     dates.update(active=False)
 
     for query_dict in [{}, {'policy__rel_date': None}]:
-        cache_key = _content_dates_cache_key(course_key, query_dict)
-        cache.delete(cache_key)
+        for subsection_and_higher_only in [False, True]:
+            for course_version in [None, published_version]:
+                cache_key = _content_dates_cache_key(
+                    course_key, query_dict, subsection_and_higher_only, course_version
+                )
+                TieredCache.delete_all_tiers(cache_key)
```

Comment:
Does it need to be called though? If the version information is encoded into the cache key?

Reply:
I would have just removed it - but the fact that a `None` published_version can still be sent in made me think that the code still should clear the cache in that case. (I also hadn't made the edx-platform PR yet which uses the new edx-when, so I wasn't sure at the time if a published course version could be passed-in for all calls. I've since found out that a course version *can* be passed-in everywhere.)

Also, in the edx-platform PR, I'm getting the published version from `course.course_version` in most places. And I don't understand completely when I can be assured that attribute will be present and filled-in. In Old Mongo courses, wouldn't that attribute be `None` or non-existent?